transactions/demo-typescript/src/tests/demo-test.ts

```diff
@@ -0,0 +1,353 @@
+import { after, before, describe, it } from "node:test";
+import * as DemoConfig from "lbf-demo-config-api/LambdaBuffers/Demo/Config.mjs";
+import * as LbrPrelude from "lbr-prelude";
+import * as LbrPlutusV1 from "lbr-plutus/V1.js";
+import * as Prelude from "prelude";
+import * as PlaV1 from "plutus-ledger-api/V1.js";
+import * as PlaPd from "plutus-ledger-api/PlutusData.js";
+import * as PreludeJson from "prelude/Json.js";
+import { EqDatum } from "lbf-demo-plutus-api/LambdaBuffers/Demo/Plutus.mjs";
+import { ChildProcess, spawn } from "node:child_process";
+import { cslPlutusDataToPlaPlutusData } from "../lib/utils.js";
+import { initRtsConfig } from "../lib/rtsconfig.js";
+import {
+  awaitTransaction,
+  createValueTx,
+  inputIsEqualTx,
+  inputIsNotEqualTx,
+  queryAddressUtxos,
+  submitTransaction,
+} from "../lib/index.js";
+import * as csl from "@emurgo/cardano-serialization-lib-nodejs";
+import * as fs from "node:fs/promises";
+import process from "node:process";
+
+/*
+ * The test suite!
+ *
+ * Notes:
+ *  - If one sets the environment variable `TEST_RTS_CONFIG`, the file
+ *    `$TEST_RTS_CONFIG` will be used in {@link initRtsConfig}.
+ *
+ *  - Otherwise, this will run the CLI command `demo-rts` (which starts all the
+ *    runtime services), and use the configuration file `demo-rts` provides to
+ *    run the show.
+ *
+ *  - In either case, we run all the tests afterwards.
+ */
+describe(`Typescript demo tests`, async () => {
+  /*
+   * Start the runtime services by executing `demo-rts` (or use the existence
+   * runtime services if `TEST_RTS_CONFIG` is set)
+   */
+  let demoRts: ChildProcess = undefined as unknown as ChildProcess;
+  const testRtsConfigFilePath = process.env[`TEST_RTS_CONFIG`];
+
+  function killDemoRts() {
+    if (demoRts.killed === false) {
+      process.kill(-demoRts.pid!, "SIGTERM");
+    }
+  }
+
+  // Start `demo-rts`
+  // NOTE(jaredponn): we use the same cluster for all tests
+  before(async () => {
+    if (testRtsConfigFilePath !== undefined) {
+      await initRtsConfig(testRtsConfigFilePath);
+    } else {
+      demoRts = spawn("demo-rts", [], {
+        stdio: ["ignore", "pipe", "ignore"],
+        detached: true,
+        env: Object.assign({ "ADA": "10" }, process.env),
+      });
```

Comment:
Ideally, I'd like for the RTS to managed (spawn, kill) from the outside, and not from the testsuite.

For example, if I point this testsuite to mainnet? You can't really spawn or kill it :D It's not urgent or really important for this PR, but generally that's where I want to go with this.